glastopf/modules/handlers/emulators/rfi.py

```diff
@@ -21,12 +21,55 @@
 import re
 import logging
 import ssl
+import socket
+import requests
+from urlparse import urlparse
+from socket import inet_aton
+from struct import unpack
+from requests.utils import requote_uri
+
+
 
 import glastopf.sandbox.sandbox as sandbox
 from glastopf.modules.handlers import base_emulator
 
 logger = logging.getLogger(__name__)
 
+def check_ssrf(url):
+    hostname = urlparse(url).hostname
+
+    def ip2long(ip_addr):
+        return unpack("!L", inet_aton(ip_addr))[0]
+
+    def is_inner_ipaddress(ip):
+        ip = ip2long(ip)
+        return ip2long('127.0.0.0') >> 24 == ip >> 24 or \
+                ip2long('10.0.0.0') >> 24 == ip >> 24 or \
+                ip2long('172.16.0.0') >> 20 == ip >> 20 or \
+                ip2long('192.168.0.0') >> 16 == ip >> 16
+
+    try:
+    	# print re.match(r"^http(s)?://(.*?)$", url)
+        # if not re.match(r"^https?://.*/.*$", url):
+        if not re.match(r"^http(s)?://(.*?)$", url):
+            raise BaseException("url format error")
+        # print socket.getaddrinfo(hostname, 'http')
+        ip_address = socket.getaddrinfo(hostname, 'http')[0][4][0]
+        if is_inner_ipaddress(ip_address):
+            logger.info("inner ip address attack")
+            raise BaseException("inner ip address attack")
```

Comment:
why is this bad?

Reply:
Because the others may attack the inner network, we must filter the url before we use our web server to request url.

Reply:
I actually like this. This should only happening when testing. Maybe instead of raising an exception give it a classification?

Reply:
Yeah, your are right.  I will change it, but nowadays I am too busy, maybe I will send a pull request slowly.